### PR TITLE
Added extract specs default

### LINKED ISSUE #261: Cluster scheduling failure with not fully registered worker nodes
## Short
When a worker node is not fully registered (e.g., handshake fails), the scheduler will fail all the scheduling tasks. 

## Proposal 
The issue happens inside `cluster-scheduler/calculation.py`
In particular here:
```
def extract_specs(node):
    return {
        'available_cpu': node.get('current_cpu_cores_free') * (100 - node.get('current_memory_percent')) / 100,
        'available_memory': node.get('current_free_memory_in_MB'),
        'available_gpu': len(node.get('gpu_info', [])),
        'virtualization': node.get('node_info').get('technology'),
    }
```

We assume all the node information is there. But when performing `node.get('node_info').get('technology')` the value of `node.get('node_info')` can be `None`

## Solution 
This can be easily solved using default values

```
        'available_cpu': node.get('current_cpu_cores_free', 0) * (100 - node.get('current_memory_percent')) / 100,
        'available_memory': node.get('current_free_memory_in_MB',0),
        'available_gpu': len(node.get('gpu_info', [])),
        'virtualization': node.get('node_info', {}).get('technology', []),
```
The node with no memory and virtualization technologies won't be considered for the scheduling.

## Status
testing

## Checklist
- [ ] Discussed
- [x] Solved
- [ ] Tested

### PR DESCRIPTION
**Pull request by giobart**
fixes #261

### CODE REVIEW

**Comment by giobart** [comment]
@Malyuk-A  This will probably integrated into your current abstractor work. Can you check this?

**Comment by Malyuk-A** [comment]
@giobart, I think there is a misunderstanding. I simply reviewed the PR of @melkodary, he is currently working on the Abstractor service. 🙂

**Comment by melkodary** [comment]
@Malyuk-A I think he tagged you by mistake. @giobart I will have a look once I get back from work.

File: cluster_orchestrator/cluster-scheduler/calculation.py [hunk 1]
```diff
@@ -100,10 +100,10 @@ def replicate(job):
 
 def extract_specs(node):
     return {
-        'available_cpu': node.get('current_cpu_cores_free') * (100 - node.get('current_memory_percent')) / 100,
-        'available_memory': node.get('current_free_memory_in_MB'),
+        'available_cpu': node.get('current_cpu_cores_free', 0) * (100 - node.get('current_memory_percent')) / 100,
+        'available_memory': node.get('current_free_memory_in_MB', 0),
         'available_gpu': len(node.get('gpu_info', [])),
-        'virtualization': node.get('node_info').get('technology'),
+        'virtualization': node.get('node_info', {}).get('technology', []),
```
Comment (melkodary) [inline cluster_orchestrator/cluster-scheduler/calculation.py:106]: Should the resource abstractor apis return: 'available_cpu/memory/gpu'? 
(Currently it returns the total_cpu/memory/gpu including some other attributes)

Comment (giobart) [reply cluster_orchestrator/cluster-scheduler/calculation.py:106]: Yes, we need both available and total.

**Pull request merged**